### PR TITLE
Fix Python bindings for Python 3.11+, move pybind11 to being a submodule

### PR DESCRIPTION
**Pull request by ConorBADI**
tl;dr - stagnant cmake files broke when Python 3.11 deprecated a configuration variable they relied on. This PR moves pybind11 to being a submodule to make future updates easier

---

Hey folks!

I'm hacking against the Python bindings at the moment, and I stumbled across the same issue as in #715 - upon building the Python bindings using Python 3.12.6 (using the latest main & `setup_project.bat`), the name of the extension module that is built is `aditofpythonNone` (instead of `aditofpython.cp312-win_amd64.pyd`, or similar).

Debugging this issue lead me to #34 and #422. In those updates, the pybind11 header files were updated, but not the supporting cmake files. As such: when the variable pybind11 gets the platform specific suffix to append to the extension file was [deprecated](https://github.com/python/cpython/issues/91670), [pybind11 pushed a fix](https://github.com/pybind/pybind11/commit/3e4d54bc0181eb7aad2ac52a2a18be18c1ab9aee), but we never received it.

In order to best keep our entire copy of pybind11 up to date, this PR moves pybind11 from an in-tree copy to a submodule. The submodule is currently set to the v2.11 branch, making the effective version bump merely a single patch version. (We currently have v2.11.1 in tree, and this PR moves us to v2.11.2 as a submodule). This is enough to catch the updates made to the cmake files in the last few years, and fix the build issues manifesting on slightly newer Python versions.

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -2,3 +2,6 @@
 	path = libaditof
 	url = https://github.com/analogdevicesinc/libaditof.git
 	branch = main
+[submodule "bindings/python/pybind11"]
+	path = bindings/python/pybind11
+	url = https://github.com/pybind/pybind11
```
Comment (SeptimiuVana) [inline .gitmodules:7]: Please specify a stable release version (probably  v2.11.2). Using main version may cause different issues.

Comment (ConorBADI) [reply .gitmodules:7]: 👍. Pinned `--remote` updates to the v2.11 branch.

**Review by SeptimiuVana — APPROVED** [review]
